Objective-C/CBLDatabase.mm

```diff
@@ -219,6 +219,60 @@ - (BOOL) saveDocument: (CBLMutableDocument*)document
 }
 
 
+- (BOOL) saveDocument: (CBLMutableDocument *)document
+      conflictHandler: (BOOL (^)(CBLMutableDocument *, CBLDocument * nullable))conflictHandler
+                error: (NSError **)error {
+    
+    CBLAssertNotNil(document);
+    CBLAssertNotNil(conflictHandler);
+    
+    while (true) {
+        NSError* err;
+        BOOL success = [self saveDocument: document
```

Comment:
When saving the document again after handling the conflict, the document should be saved on top of the oldDoc (as base).